src/responses.rs

```diff
@@ -251,9 +266,12 @@ impl EncryptionInfo {
     /// decryption key originally.
     #[wasm_bindgen(getter, js_name = "senderCurve25519Key")]
     pub fn sender_curve25519_key(&self) -> Option<JsString> {
-        Some(match &self.inner.algorithm_info {
-            AlgorithmInfo::MegolmV1AesSha2 { curve25519_key, .. } => curve25519_key.clone().into(),
-        })
+        match &self.inner.algorithm_info {
+            AlgorithmInfo::MegolmV1AesSha2 { curve25519_key, .. } => {
+                Some(curve25519_key.clone().into())
+            }
+            AlgorithmInfo::OlmV1Curve25519AesSha2 { .. } => None,
+        }
```

Comment:
again, consider copying the `curve25519_key` and `sender_claimed_ed25519_key` into the `EncryptionInfo` struct, to avoid having to do this match.

Alternatively, add a comment saying that we know the algorithm is MegolmV1AesSha2